types/address.go

```diff
@@ -72,6 +72,23 @@ func (a *Address) Decode(decoder scale.Decoder) error {
 		return fmt.Errorf("decoding of Address with 0xfe prefix not supported")
 	}
 
+	if b != 0xfd && b != 0xfc {
+		b0, err := decoder.ReadOneByte()
+		if err != nil { // Handle Index Account case
+			a.IsAccountIndex = true
```

Comment:
I moved this code from the bottom of the function. But looks like it only supports indices up to 256, is that intended even though the original type is a int32?

Reply:
@mikiquantum are you sure you just copied it? 

The logic was quite different and handled both uint16 ad uint32: https://github.com/centrifuge/go-substrate-rpc-client/blob/master/types/address.go#L59-L92

Reply:
I copied the index code from the bottom and that code looked like only supported indexes up to 256 since it only decoded 1 byte.
https://github.com/centrifuge/go-substrate-rpc-client/blob/master/types/address.go#L89-L90